models/users.js

```diff
@@ -412,6 +414,34 @@ const getUsersBasedOnFilter = async (query) => {
     const filteredUserDocs = userDocs.filter((doc) => !doc.roles?.archived);
     return filteredUserDocs;
   }
+
+  const roleQuery = query.role;
+  const verifiedQuery = query.verified;
```

Comment:
What does this verified check? If this is for checking if a particular user in out discord bot verified or not, we already do have a middleware for that. If not, then let's discuss.